### PR TITLE
feat(ClusterDropdown): authenticated cluster header label

### PR DESCRIPTION
**Pull request by weblancaster**
> Be aware this PR does not contain the styles for the header (@tnguyen0 has been working on it).

Commit adds authenticated cluster dropdown label mechanism
to display the correct info when metadata available to the user
or to display the hardcoded string.

Closes DCOS-39679

## Testing

1. `npm start`
2. Go to `HeaderBar.tsx` comment out line 12 - 14.
3. Check out corresponding branch on privates plugin

**Case 1:**
User has admin permissions (access to metadata): user should see the cluster name and IP on the header.
jul-31-2018 16-41-41

**Case 2**:
User doesn't have access to metadata = Cluster dropdown on Header displays `cluster` string only.
jul-31-2018 16-40-16

**How to create a user without metadata access**
jul-31-2018 17-55-40

### CODE REVIEW

File: plugins/oauth/components/AuthenticatedClusterDropdown.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import ClusterDropdown from "#SRC/js/components/ClusterDropdown";
+
+class AuthenticatedClusterDropdown extends ClusterDropdown {
```
Comment (juliangieseke) [inline plugins/oauth/components/AuthenticatedClusterDropdown.js]: is it okay to extend react components? AFAIK react doesnt really support this pattern 🤔

File: src/js/components/HeaderBar.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import * as React from "react";
 
 import { Link } from "react-router";
 import { HeaderBar as UIHeaderBar } from "ui-kit-stage/HeaderBar";
-import ClusterDropdown from "./ClusterDropdown";
+import ClusterHeader from "./ClusterHeader";
```
Comment (juliangieseke) [inline src/js/components/HeaderBar.tsx]: why the rename? 🤔

**Comment by juliangieseke** [comment]
^ and it is blocking EE PR…

**Comment by nLight** [comment]
Let's please _not rush_ the merges even if they are blocking something when they are NOT of a quality.

File: plugins/oauth/hooks.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ import CookieUtils from "#SRC/js/utils/CookieUtils";
 import RouterUtil from "#SRC/js/utils/RouterUtil";
 import UsersPage from "#SRC/js/pages/system/UsersPage";
 
-import AuthenticatedUserAccountDropdown from "./components/AuthenticatedUserAccountDropdown";
+import AuthenticatedClusterDropdown from "./components/AuthenticatedClusterDropdown";
```
Comment (tnguyen0) [inline plugins/oauth/hooks.js]: The changes in these file are therefore not needed.

Comment (weblancaster) [reply plugins/oauth/hooks.js]: Discussed offline.. we need the file.

**Pull request merged**